server/database.js

```diff
@@ -0,0 +1,65 @@
+const {getDomain} = require('./domain.js');
+
+//Endpoint to handle requests to access participant data
+//queries databse for a specific url corresponding to a certain group 
+//of users and returns all rows in the evidence table with that url
+//Returns 200 unless there was an error with the database query
+function dataEndpoint(pool, request, response) {
+  const token = request.headers['x-teachermoments-token'];
+  const domain = getDomain(request);
+  const location = `${domain}/teachermoments/turner?KevinTesting20180319`;
+
+  checkAccess(pool, location, token)
+    .then(isTokenAuthorized => {
+      if (isTokenAuthorized) {
+        getData(pool, location)
+          .then(results => {
+            response.set('Content-Type', 'application/json');
+            response.json({
+              evidence: {rows: results} 
+            });
+            return response.status(200).end();
+          });
+      }
+      else{
+        console.log('Researcher is not authorized to see this data');
+        return response.status(405).end();
+      }
+    })
+    .catch(err => {
+      console.log('Error with database query:', err);
+      return response.status(500).end();
+    });
+}
+
+// Spoofing the part where I confirm a token is allowed to view data from 
+// a specific location. Location is currently hard coded so this is fine.
+function checkAccess(pool, location, token) {
```

Comment:
It's a good idea to explain this and your comments here are great.  But having a `checkAccess` function that doesn't actually check access can only lead to mistakes that mistakenly grants access to user data.

One thing you could do here is add in `location === `${domain}/teachermoments/turner?KevinTesting20180319` again in here.  That way you'd hard-code that the access check only allows access to one particular location.

Reply:
You could also move this into `getData` so that a developer can't even get the data without passing the `token`, and then `getData` always calls `checkAccess`.

Reply:
gotcha. I added in both of your suggestions! Will look into fixing all of this temporary stuff after Friday!